src/main/java/mate/academy/bookshop/service/impl/BookServiceImpl.java

```diff
@@ -23,13 +25,21 @@ public class BookServiceImpl implements BookService {
 
     @Override
     public BookDto save(BookRequestDto createBookRequestDto) {
+        if (existByIsbn(IsbnFormatter.format(createBookRequestDto.getIsbn()))) {
+            throw new NotUniqueValueException("ISBN must be unique");
+        }
         Book book = bookMapper.toModel(createBookRequestDto);
         return bookMapper.toDto(bookRepository.save(book));
     }
 
     @Override
     public BookDto update(Long id, BookRequestDto updateBookRequestDto) {
-        findById(id);
+        BookDto createdBook = findById(id);

```

Comment:
we did not create it here
```suggestion
        BookDto existingBook = findById(id);
```